lightning_client_test.go

```diff
@@ -0,0 +1,182 @@
+	m.addInvoiceArgs = append(m.addInvoiceArgs, addInvoiceArg{
+		in:   in,
+		opts: opts,
+	})
+
+	return m.addInvoice(in, opts...)
+}
+
+func Test_lightningClientAddInvoice(t *testing.T) {
+	// Set up the test structure.
+	type fields struct {
+		adminMac serializedMacaroon
+		client   mockRPCClient
+		timeout  time.Duration
+	}
+
+	type args struct {
+		in *invoicesrpc.AddInvoiceData
+	}
+
+	type expect struct {
+		addInvoiceArgs []addInvoiceArg
+		hash           lntypes.Hash
+		payRequest     string
+		wantErr        bool
+	}
+
+	type testCase struct {
+		name   string
+		fields fields
+		args   args
+		expect expect
+	}
+
+	// Define constants / fixtures.
+	var validPreimage lntypes.Preimage
+	copy(validPreimage[:], "valid preimage")
+	var validRHash lntypes.Hash
+	copy(validRHash[:], "valid hash")
+	validAddInvoiceData := &invoicesrpc.AddInvoiceData{
+		Memo:            "fake memo",
+		Preimage:        &validPreimage,
+		Hash:            &validRHash,
+		Value:           lnwire.MilliSatoshi(500000),
+		DescriptionHash: []byte("fake 32 byte hash"),
+		Expiry:          123,
+		CltvExpiry:      456,
+	}
+
+	validInvoice := &lnrpc.Invoice{
+		Memo:            validAddInvoiceData.Memo,
+		RPreimage:       validAddInvoiceData.Preimage[:],
+		RHash:           validAddInvoiceData.Hash[:],
+		ValueMsat:       int64(validAddInvoiceData.Value),
+		DescriptionHash: validAddInvoiceData.DescriptionHash,
+		Expiry:          validAddInvoiceData.Expiry,
+		CltvExpiry:      validAddInvoiceData.CltvExpiry,
+		Private:         true,
+	}
+
+	validPayReq := "a valid pay req"
+	validResp := &lnrpc.AddInvoiceResponse{
+		RHash:          validRHash[:],
+		PaymentRequest: validPayReq,
+	}
+
+	validAddInvoiceArgs := []addInvoiceArg{
+		{in: validInvoice},
+	}
+
+	validAddInvoice := func(in *lnrpc.Invoice, opts ...grpc.CallOption) (
+		*lnrpc.AddInvoiceResponse, error) {
+		return validResp, nil
+	}
+
+	errorAddInvoice := func(in *lnrpc.Invoice, opts ...grpc.CallOption) (
+		*lnrpc.AddInvoiceResponse, error) {
+		return nil, errors.New("error")
+	}
+
+	// Run through the test cases.
+	for _, tc := range []testCase{
+		{
+			name: "happy path",
+			fields: fields{
+				client: mockRPCClient{
+					addInvoice: validAddInvoice,
+				},
+			},
+			args: args{
+				in: validAddInvoiceData,
+			},
+			expect: expect{
+				addInvoiceArgs: validAddInvoiceArgs,
+				hash:           validRHash,
+				payRequest:     validPayReq,
+			},
+		}, {
+			name: "rpc client error",
+			fields: fields{
+				client: mockRPCClient{
+					addInvoice: errorAddInvoice,
+				},
+			},
+			args: args{
+				in: validAddInvoiceData,
+			},
+			expect: expect{
+				addInvoiceArgs: validAddInvoiceArgs,
+				wantErr:        true,
+			},
+		},
+	} {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			ln := lightningClient{
+				client:   &tc.fields.client,
+				timeout:  tc.fields.timeout,
+				adminMac: tc.fields.adminMac,
+			}
+
+			hash, payRequest, err := ln.AddInvoice(
```

Comment:
nit: Multi-line method calls should be formatted as:

```
hash, payRequest, err := ln.AddInvoice(
    context.Background(), tc.args.in,
)
```

Same with the `require.Equal` calls below.

Reply:
Updated with most recent commit – let me know if not what you had in mind, though. ⚡